kafka_utils/util/ssh.py

```diff
@@ -0,0 +1,158 @@
+# -*- coding: utf-8 -*-
+# Copyright 2016 Yelp Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from __future__ import absolute_import
+from __future__ import print_function
+from __future__ import unicode_literals
+
+import socket
+import sys
+import time
+from contextlib import closing
+from contextlib import contextmanager
+
+from paramiko.agent import AgentRequestHandler
+from paramiko.client import AutoAddPolicy
+from paramiko.client import SSHClient
+
+from kafka_utils.util.error import MaxConnectionAttemptsError
+
+
+class Connection:
+    """Represents a SSH connection with an SSH server.
+    """
+
+    def __init__(self, client, forward_agent, sudoable):
+        self.transport = client.get_transport()
+        self.forward_agent = forward_agent
+        self.sudoable = sudoable
+
+    def sudo_command(self, command, bufsize=-1):
+        """Sudo a command on the SSH server.
+        Delegates to :func`~ssh.Connection.exec_command`
+
+        :param command: the command to execute
+        :type command: str
+        :param bufsize: interpreted the same way as by the built-in C{file()} function in python
+        :type bufsize: int
+        :returns the stdin, stdout, and stderr of the executing command
+        :rtype: tuple(L{ChannelFile}, L{ChannelFile}, L{ChannelFile})
+
+        :raises SSHException: if the server fails to execute the command
+        """
+        new_command = "sudo -S -p 'sudo password:' {0}".format(command)
```

Comment:
@ecanzonieri Should I just replace sudo_command and just use exec_command? As long as the connection is sudoable, we can just prefix the commands with sudo.